### PR TITLE
[ADAM-1616] Resolve installation directory if scripts are symlinks.

### LINKED ISSUE #1616: adam-submit cannot find assembly jar if installed as symlink
For better or worse, the bioconda guidelines are to install wrappers such as `adam-submit` and `adam-shell` as symlinks.

https://bioconda.github.io/guidelines.html#java

Our wrapper scripts cannot find the assembly jar if installed this way.

```
...
$ mkdir noarch
$ echo '{}' > noarch/repodata.json
$ bzip2 -k noarch/repodata.json
+ source /opt/conda/bin/activate /opt/conda/conda-bld/adam_1500624861141/_t_env
+ /bin/bash -x -e /opt/conda/conda-bld/adam_1500624861141/test_tmp/run_test.sh
+ adam-submit --help
ls: cannot access /opt/conda/conda-bld/adam_1500624861141/_t_env/adam-assembly/target: No such file or directory
Failed to find ADAM cli assembly in /opt/conda/conda-bld/adam_1500624861141/_t_env/adam-assembly/target.
You need to build ADAM before running this program.
```

### PR DESCRIPTION
**Pull request by heuermh**
Fixes #1616

### CODE REVIEW

**Comment by heuermh** [comment]
Jenkins, retest this please.

**Comment by heuermh** [comment]
Jenkins, retest this please.

File: bin/adam-shell [hunk 1]
```diff
@@ -19,21 +19,29 @@
...
-# Figure out where ADAM is installed
-SCRIPT_DIR="$(cd `dirname $0`/..; pwd)"
-
 # does the user have ADAM_OPTS set? if yes, then warn
 if [[ -z $@ && -n "$ADAM_OPTS" ]]; then
     echo "WARNING: Passing Spark arguments via ADAM_OPTS was recently removed."
     echo "Run adam-shell instead as adam-shell <spark-args>"
 fi
 
+# Find original directory of this script, resovling symlinks
```
Comment (fnothaft) [inline bin/adam-shell]: `resovling` -> `resolving`

File: bin/adam-submit [hunk 2]
```diff
@@ -49,15 +49,23 @@ if [[ $DD == False && -n "$ADAM_OPTS" ]]; then
     echo "Run adam-submit instead as adam-submit <spark-args> -- <adam-args>"
 fi
 
-# Figure out where ADAM is installed
-SCRIPT_DIR="$(cd `dirname $0`/..; pwd)"
+# Find original directory of this script, resovling symlinks
```
Comment (fnothaft) [inline bin/adam-submit]: `resovling` -> `resolving`

**Comment by fnothaft** [comment]
Jenkins, test this please.

**Comment by fnothaft** [comment]
Merged manually as e3fa6bfc9af9f27863224fcb6008d4eaed879eff. Thanks @heuermh!

**Comment by heuermh** [comment]
Thank you!

**Pull request closed without merging**